components/org.wso2.carbon.identity.api.server.flow.management/org.wso2.carbon.identity.api.server.flow.management.v1/src/main/java/org/wso2/carbon/identity/api/server/flow/management/v1/response/handlers/AbstractMetaResponseHandler.java

```diff
@@ -0,0 +1,133 @@
+import org.wso2.carbon.identity.api.server.flow.management.v1.utils.Utils;
+import org.wso2.carbon.identity.application.common.model.IdentityProvider;
+import org.wso2.carbon.identity.multi.attribute.login.constants.MultiAttributeLoginConstants;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Abstract class for handling meta responses for different flows.
+ * This class provides methods to retrieve flow type, attribute profile,
+ * supported executors, connector configurations, and connection meta information.
+ */
+public abstract class AbstractMetaResponseHandler {
+
+    /**
+     * Get the flow type.
+     *
+     * @return Flow type.
+     */
+    public abstract String getFlowType();
+
+    /**
+     * Get the attribute profile.
+     *
+     * @return Attribute profile.
+     */
+    public abstract String getAttributeProfile();
+
+    /**
+     * Get the list of supported executors for the flow.
+     *
+     * @return List of supported executors.
+     */
+    public abstract List<String> getSupportedExecutors();
+
+    /**
+     * Get the connector configurations for the flow.
+     *
+     * @return Connector configurations.
+     */
+    public abstract BaseConnectorConfigs getConnectorConfigs();
+
+    /**
+     * Get the required input fields for the flow.
+     *
+     * @return List of required input fields.
+     */
+    public abstract List<String> getRequiredInputFields();
+
+    /**
+     * Create the meta response object populated with common values.
+     *
+     * @return BaseFlowMetaResponse instance.
+     */
+    public BaseFlowMetaResponse createResponse() {
+
+        BaseFlowMetaResponse response = new BaseFlowMetaResponse();
+        response.setFlowType(getFlowType());
+        response.setAttributeProfile(getAttributeProfile());
+        response.setSupportedExecutors(getSupportedExecutors());
+        response.setConnectorConfigs(getConnectorConfigs());
+        return response;
+    }
+
+    /**
+     * Return login related input fields common across flows.
+     */
+    protected List<String> getLoginInputFields() {
+
+        Utils utils = new Utils();
+        String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        List<String> fields = new java.util.ArrayList<>();
+        if (utils.isFlowConfigEnabled(tenantDomain, MultiAttributeLoginConstants.MULTI_ATTRIBUTE_LOGIN_PROPERTY)) {
+            fields.add(FlowEndpointConstants.USER_IDENTIFIER);
+            fields.add(FlowEndpointConstants.USERNAME_IDENTIFIER);
+        } else {
+            fields.add(FlowEndpointConstants.USERNAME_IDENTIFIER);
+        }
+        return fields;
+    }
+
+    /**
+     * Get the connection meta information for the flow.
+     *
+     * @return Connection meta information.
+     */
+    public FlowMetaResponseConnectionMeta getConnectionMeta() {
```

Comment:
Shall we move this method into the self-registration meta response handler, since it's only required for that?

Reply:
Make it private as well